torch/utils/checkpoint.py

```diff
@@ -0,0 +1,114 @@
+import torch
+
+
+def detach_variable(inputs):
```

Comment:
I think this function doesn't need to be recursive. You can do the same thing with a tuple comprehension, because you only pass `ctx.saved_variables` into it, and it's guaranteed to be `Tuple[Tensor]`

Reply:
done